### PR TITLE
chore(v2): fix several lint warnings, add missing types, cleanup

### PR DESCRIPTION
**Pull request by Simek**
## Motivation

This is cleanup PR which includes the following changes:
* ~50 lint warnings fixed (current count is at 212)
* bunch of missing type declaration added
* missing types dependencies added
* few typos fixed

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Build & run of Docusaurs 2 website locally.

## Related PRs

No.

### CODE REVIEW

File: packages/docusaurus-mdx-loader/src/remark/transformImage/index.js [hunk 1]
```diff
@@ -18,21 +18,22 @@ const {
 } = getFileLoaderUtils();
 
 const createJSX = (node, pathUrl) => {
-  node.type = 'jsx';
-  node.value = `<img ${node.alt ? `alt={"${escapeHtml(node.alt)}"} ` : ''}${
+  const jsxNode = node;
```
Comment (slorber) [inline packages/docusaurus-mdx-loader/src/remark/transformImage/index.js:21]: why not rename the fn param directly?

File: packages/docusaurus-cssnano-preset/src/remove-overridden-custom-properties/index.js [hunk 2]
```diff
@@ -28,11 +28,13 @@ module.exports = postcss.plugin(
...
         const hasImportantProperties = sameProperties.some((p) =>
-          p.hasOwnProperty('important'),
+          Object.prototype.hasOwnProperty.call(p, 'important'),
         );
 
         const overriddenProperties = hasImportantProperties
-          ? sameProperties.filter((p) => !p.hasOwnProperty('important'))
+          ? sameProperties.filter(
+              (p) => !Object.prototype.hasOwnProperty.call(p, 'important'),
+            )
```
Comment (slorber) [inline packages/docusaurus-cssnano-preset/src/remove-overridden-custom-properties/index.js:37]: that feels a bit verbose, wonder if we shouldn't just disable this eslint rule instead?

File: packages/docusaurus-migrate/src/index.ts [hunk 3]
```diff
@@ -476,7 +477,7 @@ function handleVersioning(
     const loadedVersions: Array<string> = JSON.parse(
       String(fs.readFileSync(path.join(siteDir, 'versions.json'))),
     );
-    fs.copyFile(
+    fs.copyFileSync(
```
Comment (slorber) [inline packages/docusaurus-migrate/src/index.ts:480]: good catch :D

File: packages/docusaurus-plugin-sitemap/src/pluginOptionSchema.ts [hunk 4]
```diff
@@ -5,19 +5,20 @@
...
+  changefreq: EnumChangefreq.WEEKLY,
   priority: 0.5,
   trailingSlash: false,
 };
 
 export const PluginOptionSchema = Joi.object({
   cacheTime: Joi.number().positive().default(DEFAULT_OPTIONS.cacheTime),
   changefreq: Joi.string()
-    .valid('always', 'hourly', 'daily', 'weekly', 'monthly', 'yearly', 'never')
+    .valid(...Object.values(EnumChangefreq))
```
Comment (slorber) [inline packages/docusaurus-plugin-sitemap/src/pluginOptionSchema.ts:21]: not very fan of enums (non-std js), and not alone
https://fettblog.eu/tidy-typescript-avoid-enums/
https://www.techatbloomberg.com/blog/10-insights-adopting-typescript-at-scale/

I think we'd rather avoid using them when possible, even if the underlying lib does use them

**Review by slorber — CHANGES_REQUESTED** [review]
Thanks

Most of this looks like, but we'd rather avoid TS enums, and hasOwnProperty is probably fine to use?

File: packages/docusaurus-mdx-loader/src/remark/transformImage/index.js (hunk 1 above)
Comment (Simek) [reply packages/docusaurus-mdx-loader/src/remark/transformImage/index.js:21]: `no-param-reassign` rule - https://eslint.org/docs/rules/no-param-reassign

File: packages/docusaurus-migrate/src/index.ts (hunk 3 above)
Comment (Simek) [reply packages/docusaurus-migrate/src/index.ts:480]: I should thank to static code analysis build-in IDE 😉 There was a warning about unhandled `Promise`.

File: packages/docusaurus-plugin-sitemap/src/pluginOptionSchema.ts (hunk 4 above)
Comment (Simek) [reply packages/docusaurus-plugin-sitemap/src/pluginOptionSchema.ts:21]: Yeah, I also not a big fan of Enums, but the [String variety](https://www.typescriptlang.org/docs/handbook/enums.html#string-enums) isn't that bad as for example, homogenous Enums.

Of course I can also change back the types. Would you like to revert `EnumChangefreq` usage completely or only the Enum values spread?

File: packages/docusaurus-cssnano-preset/src/remove-overridden-custom-properties/index.js (hunk 2 above)
Comment (Simek) [reply packages/docusaurus-cssnano-preset/src/remove-overridden-custom-properties/index.js:37]: I think it's fair to disable it, as much as leaving it on. The side effects or DDoS issues mentioned [on the rule page](https://eslint.org/docs/rules/no-prototype-builtins) seems not be related to the way those methods are used in the Docusuaurs code, but in the other hand I think that the slightly more verbose syntax won't hurt anyone, and there are only a few usages of `hasOwnProperty` in the wholde codebase.

File: packages/docusaurus-plugin-sitemap/src/pluginOptionSchema.ts (hunk 4 above)
Comment (slorber) [reply packages/docusaurus-plugin-sitemap/src/pluginOptionSchema.ts:21]: let's keep it, just hope libs will stop exporting enums in the future :)

**Pull request merged**